### PR TITLE
Add Portal module with list_organizations function

### PR DESCRIPTION
**Pull request by willmanduffy**
* Returns a list of the organizations accessible to the provided
  API key. Includes pagination functionality via the list_metadata
  property.

### CODE REVIEW

File: tests/test_directory_sync.py
Comment (willmanduffy) [inline tests/test_directory_sync.py:11]: Small unrelated change. I noticed this typo while I was copying other people's code :)

**Review by maxchehab — APPROVED** [review]
lgtm, but I'll defer to @henrylamchan or @rjvani for the 👍

File: workos/portal.py
Comment (henrylamchan) [inline workos/portal.py:24]: Kwargs not Args

File: workos/portal.py
Comment (henrylamchan) [inline workos/portal.py:20]: Looks like the params expects on API are domains (plural not singular). Does this work as is?

**Review by henrylamchan — CHANGES_REQUESTED** [review]

File: workos/portal.py
Comment (willmanduffy) [reply workos/portal.py:20]: This is a great catch @henrylamchan looks like my original spec was wrong and I tracked this bug across multiple SDKs. I'll fix here and give a fix to Node and Ruby before releasing them. Good eye!

File: workos/portal.py
Comment (henrylamchan) [inline workos/portal.py:34]: I think a gotcha that I ran into when implementing get events is that the way query params are serialized from the python requests library.. is valid but nestjs throws up on validation because a list with one item is deserialized as just a string. that being said, you may want to name the key for `domains` as `domains[]` because of that

Comment (willmanduffy) [reply workos/portal.py:34]: Hmmm weird, I did a full request to the API and it seemed to handle it just fine. Happy to swap this up though!

Comment (henrylamchan) [reply workos/portal.py:34]: Ah, if it works its all good /shrug

Comment (willmanduffy) [reply workos/portal.py:34]: Just double checked. I added a real API and did this diff:

```diff
     def test_list_organizations(self, mock_organizations, mock_request_method):
-        mock_response = Response()
-        mock_response.status_code = 200
-        mock_response.response_dict = mock_organizations
-        mock_request_method("get", mock_response, 200)
-        response = self.portal.list_organizations()
-        assert response.status_code == 200
-        assert len(response.response_dict["data"]) == 2
+        response = self.portal.list_organizations(domains=["example2.com"])
+        assert len(response["data"]) == 1
+        assert response["data"] == True
```

First test passes and second test fails but shows the response I expected. Feeling good it works so I'm going to dismiss the blocking review and merge!

**Review by henrylamchan — APPROVED** [review]

**Pull request merged**
